Dockerfile

```
@@ -12,21 +16,45 @@ RUN apk add --no-cache \
 COPY package*.json ./
-RUN npm ci --only=production
+# --omit=dev replaces the deprecated --only=production (removed in npm v12).
+RUN npm ci --omit=dev && npm cache clean --force
+
+# ---- runtime stage -----------------------------------------------------------
+FROM node:22-alpine
+
+# Shared libraries canvas links against at runtime (the -dev headers and the
+# compiler are deliberately left behind in the build stage), plus the DejaVu
+# fonts that utils/shopBanner.js and utils/cardGenerator.js register by path.
+RUN apk add --no-cache \
```

Hadolint (2.14.0)

[warning] Pin versions in apk add. Instead of `apk add <package>` use `apk add <package>=<version>`

(DL3018)